buildSrc/src/main/java/org/opensearch/gradle/zipplugin/ZipPublish.java

```diff
@@ -0,0 +1,122 @@
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+import java.nio.file.Path;
+import org.gradle.api.Task;
+
+public class ZipPublish implements Plugin<Project> {
+    private Project project;
+
+    public final static String EXTENSION_NAME = "zipmavensettings";
+    public final static String PUBLICATION_NAME = "mavenzip";
+    public final static String STAGING_REPO = "zipstaging";
+    public final static String MAVEN_ZIP_PUBLISH_TASK = "publish"
+        + ZipPublishUtil.capitalize(PUBLICATION_NAME)
+        + "PublicationTo"
+        + ZipPublishUtil.capitalize(STAGING_REPO)
+        + "Repository";
+    public final static String MAVEN_ZIP_PUBLISH_POM_TASK = "generatePomFileFor"
+        + ZipPublishUtil.capitalize(PUBLICATION_NAME)
+        + "Publication";
+    public final static String LOCALMAVEN = "publishToMavenLocal";
+    public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
+    public static String BUILD_DISTRIBUTIONS_LOCATION = "/build/distributions/";
+
+    private void configMaven() {
+        final Path buildDirectory = this.project.getRootDir().toPath();
+        this.project.getPluginManager().apply(MavenPublishPlugin.class);
+        this.project.getExtensions().configure(PublishingExtension.class, publishing -> {
+            publishing.repositories(repositories -> {
+                repositories.maven(maven -> {
+                    maven.setName(STAGING_REPO);
+                    maven.setUrl(buildDirectory.toString() + LOCAL_STAGING_REPO_PATH);
+                });
+            });
+            System.out.println("Starting " + MAVEN_ZIP_PUBLISH_TASK + " task");
+            publishing.publications(publications -> {
+                publications.create(PUBLICATION_NAME, MavenPublication.class, mavenZip -> {
+                    ZipPublishExtension extset = this.project.getExtensions().findByType(ZipPublishExtension.class);
+                    // Getting the Zip group from created extension
+                    String zipGroup = extset.getZipgroup();
+                    String zipArtifact = getProperty("zipArtifact");
+                    // Getting the Zip version from gradle property with/without added snapshot and qualifier
+                    String zipVersion = System.getProperty("opensearch.version");
+                    String version = null;
+                    String extraSuffix = null;
+                    if (zipVersion != null) {
+                        StringTokenizer st = new StringTokenizer(zipVersion);
+                        version = st.nextToken("-") + ".0";
+                        try {
+                            extraSuffix = zipVersion.substring(zipVersion.indexOf("-"));
+                        } catch (Exception e) {
+                            System.out.println("");
+                        }
+                    }
+                    String finalZipVersion = version + extraSuffix;
```

Comment:
This is backwards. You have access to version and qualifier properties, so the last thing you should be doing is parsing the ZIP filename. Build the correct version from those.

Reply:
Hey @dblock, the `$VERSION` currently already has the qualifier and snapshot info ex `2.0.0-rc1-SNAPSHOT`, it will save some flags for running the task and would reduce the human errors
Now a user just need to pass `./gradlew publishMavenzipPublicationToZipstagingRepository -Dopensearch.version=$VERSION` , dont need to pass any additional flags.

Reply:
The version that is getting passed has the value as `VERSION=2.0.0-alpha1-SNAPSHOT`, not just `2.0.0`, hence I went with that parsing logic as in `$VERSION` itself I can get the qualifier and snapshot info. 
so now a user need not pass all other flags `version_qualifier` and `build.snapshot`. I can try with `VersionProperties.getOpenSearch()`, but again it this only gives just the version number, then we expect a user to add the other flags to execute the task.

Reply:
from the existing task `bundlePlugin` coming from [PluginBuildPlugin](https://github.com/opensearch-project/OpenSearch/blob/05a58192433f6e007ade61b4b4f21d0f068d50e0/buildSrc/src/main/java/org/opensearch/gradle/plugin/PluginPropertiesExtension.java), the output is an generated as a zip file, I have added the publish artifact to look for this output.

Reply:
The `zipArtifact` name will be the project root name and `version` is the gradle property of the project.